### PR TITLE
DM: update openapi spec

### PR DESCRIPTION
**Pull request by Ehco1996**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tiflow/pull/4627

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [x] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by shichun-0415** [comment]
@Ehco1996 Could you involve a technical reviewer? Thanks.

File: dm/dm-open-api.md [hunk 1]
```diff
@@ -248,6 +258,211 @@ curl -X 'POST' \
...
+  },
+  "purge": {
+    "interval": 3600,
+    "expires": 0,
+    "remain_space": 15
+  }
+}
+```
+
+## 开启数据源
```
Comment (sunzhaoyang) [inline dm/dm-open-api.md:428]:
```suggestion
## 启用数据源
```

Comment (sunzhaoyang) [reply dm/dm-open-api.md:428]: 我觉得 开启对应 start，enable 用启用比较好。

File: dm/dm-open-api.md [hunk 2]
```diff
@@ -248,6 +258,211 @@ curl -X 'POST' \
...
+{{< copyable "shell-regular" >}}
+
+```shell
+curl -X 'POST' \
+  'http://127.0.0.1:8261/api/v1/sources/mysql-01/enable' \
+  -H 'accept: */*' \
+  -H 'Content-Type: application/json'
+```
+
+## 关闭数据源
```
Comment (sunzhaoyang) [inline dm/dm-open-api.md:447]:
```suggestion
## 停用数据源
```

File: dm/dm-open-api.md [hunk 3]
```diff
@@ -248,6 +258,211 @@ curl -X 'POST' \
...
+
+这是一个同步接口，请求成功的时会批量停止当前数据源上运行的所有同步任务。
```
Comment (sunzhaoyang) [inline dm/dm-open-api.md:449]:
```suggestion
这是一个同步接口，请求成功的时会停用此数据源，并批量停止数据迁移任务中依赖该数据源的所有子任务。
```

File: dm/dm-open-api.md [hunk 4]
```diff
@@ -248,6 +258,211 @@ curl -X 'POST' \
...
+    "interval": 3600,
+    "expires": 0,
+    "remain_space": 15
+  }
+}
+```
+
+## 开启数据源
+
+这是一个同步接口，请求成功的时会批量开启当前数据源上运行的所有同步任务。
```
Comment (sunzhaoyang) [inline dm/dm-open-api.md:430]:
```suggestion
这是一个同步接口，请求成功的时会启用此数据源，并批量开始数据迁移任务中依赖该数据源的所有子任务。```

File: dm/dm-open-api.md [hunk 5]
```diff
@@ -410,40 +607,27 @@ curl -X 'POST' \
...
-{{< copyable "shell-regular" >}}
-
-```shell
-curl -X 'POST' \
-  'http://127.0.0.1:8261/api/v1/sources/mysql-01/pause-relay' \
-  -H 'accept: */*'
-```
-
-## 对数据源恢复 relay-log 功能
+## 清除对数据源上多余的relay-log文件
```
Comment (sunzhaoyang) [inline dm/dm-open-api.md:610]:
```suggestion
## 清除对数据源上多余的 relay log 文件
```

File: dm/dm-open-api.md [hunk 6]
```diff
@@ -716,133 +896,364 @@ curl -X 'POST' \
...
+  -H 'accept: application/json'
+```
+
+## 更新同步任务
+
+该接口是一个同步接口，请求成功会返回对应数据源信息。
+
+> **注意：**
+>
+> 更新同步任务配置时需要该任务处于暂停状态并已经运行到增量同步的阶段，另外只有部分字段可以更新。
```
Comment (sunzhaoyang) [inline dm/dm-open-api.md:1030]:
```suggestion
> 更新同步任务配置时需要该任务处于暂停状态并已经运行到增量同步的阶段，且仅有部分字段可以更新。
```

**Review by sunzhaoyang — APPROVED** [review]
rest LGTM

File: dm/dm-open-api.md [hunk 7]
```diff
@@ -716,133 +896,364 @@ curl -X 'POST' \
 }
 ```
 
-## 获取同步任务列表
+## 获取同步任务
 
-该接口是一个同步接口，请求成功会返回对应的同步任务信息。
+这是一个同接口，请求成功的 Status Code 是 200
```
Comment (Ehco1996) [inline dm/dm-open-api.md:901]:
```suggestion
这是一个同步接口，请求成功的 Status Code 是 200
```

File: dm/dm-open-api.md [hunk 8]
```diff
@@ -716,133 +896,364 @@ curl -X 'POST' \
...
+
+```shell
+curl -X 'DELETE' \
+  'http://127.0.0.1:8261/api/v1/tasks/task-1' \
+  -H 'accept: application/json'
+```
+
+## 更新同步任务
+
+该接口是一个同步接口，请求成功会返回对应数据源信息。
```
Comment (Ehco1996) [inline dm/dm-open-api.md:1026]:
```suggestion
该接口是一个同步接口，请求成功会返回对应同步任务的信息。
```

File: dm/dm-open-api.md [hunk 9]
```diff
@@ -716,133 +896,364 @@ curl -X 'POST' \
...
+  -H 'accept: application/json'
+```
+
+## 更新同步任务
+
+该接口是一个同步接口，请求成功会返回对应同步任务的信息。
+
+> **注意：**
+>
+> 更新同步任务配置时需要该任务处于暂停状态并已经运行到增量同步的阶段，且仅有部分字段可以更新。
```
Comment (TomShawn) [inline dm/dm-open-api.md:1030]:
```suggestion
> 更新同步任务配置时，须确保该任务处于暂停状态，并已经运行到增量同步的阶段，且仅有部分字段可以更新。
```

File: dm/dm-open-api.md [hunk 10]
```diff
@@ -716,133 +896,364 @@ curl -X 'POST' \
...
+        "binlog_gtid": "03fc0263-28c7-11e7-a653-6c0b84d59f30:1-7041423,05474d3c-28c7-11e7-8352-203db246dd3d:1-170"
+      }
+    ]
+  }
 }
 ```
 
+## 开始同步任务
+
+这是一个异步接口，请求成功的 Status Code 是 200，可通过[获取同步任务状态](#获取同步任务状态)接口获取最新的状态。
```
Comment (TomShawn) [inline dm/dm-open-api.md:1226]:
```suggestion
这是一个异步接口，请求成功的 Status Code 是 200。可通过[获取同步任务状态](#获取同步任务状态)接口获取最新的任务状态。
```

File: dm/dm-open-api.md [hunk 11]
```diff
@@ -716,133 +896,364 @@ curl -X 'POST' \
...
+```shell
+curl -X 'POST' \
+  'http://127.0.0.1:8261/api/v1/tasks/task-1/start' \
+  -H 'accept: */*'
+```
+
 ## 停止同步任务
 
-这是一个异步接口，请求成功的 Status Code 是 204，可通过[获取同步任务状态](#获取同步任务状态)接口获取最新的状态。
+这是一个异步接口，请求成功的 Status Code 是 200，可通过[获取同步任务状态](#获取同步任务状态)接口获取最新的状态。
```
Comment (TomShawn) [inline dm/dm-open-api.md:1244]:
```suggestion
这是一个异步接口，请求成功的 Status Code 是 200，可通过[获取同步任务状态](#获取同步任务状态)接口获取最新的任务状态。
```

File: dm/dm-open-api.md [hunk 12]
```diff
@@ -917,48 +1328,148 @@ curl -X 'GET' \
...
+          }
+        ]
+      }
+    }
+  ]
+}
 ```
 
-## 恢复同步任务
+## 获取同步任务同步规则列表
```
Comment (TomShawn) [inline dm/dm-open-api.md:1443]:
```suggestion
## 获取同步任务的同步规则列表
```

File: dm/dm-open-api.md [hunk 13]
```diff
@@ -917,48 +1328,148 @@ curl -X 'GET' \
...
 
-这是一个异步接口，请求成功的 Status Code 是 200，可通过[获取同步任务状态](#获取同步任务状态)接口获取最新的状态。
+该接口是一个同步接口，请求成功会返回对应的列表。
```
Comment (TomShawn) [inline dm/dm-open-api.md:1445]:
```suggestion
该接口是一个同步接口，请求成功会返回对应同步任务的同步列表。
```

**Review by TomShawn — COMMENTED** [review]
all minor comments

File: dm/dm-open-api.md [hunk 14]
```diff
@@ -5,17 +5,29 @@ summary: 了解如何使用 OpenAPI 接口来管理集群状态和数据同步
...
+    openapi: true
+```
+
 > **注意：**
 >
 > - DM 提供符合 OpenAPI 3.0.0 标准的 [Spec 文档](https://github.com/pingcap/tiflow/blob/master/dm/openapi/spec/dm.yaml)，其中包含了所有 API 的请求参数和返回体，你可自行复制到如 [Swagger Editor](https://editor.swagger.io/) 等工具中在线预览文档。
 >
 > - 部署 DM-master 后，你可访问 `http://{master-addr}/api/v1/docs` 在线预览文档。
+>
+> - 相比于 v5.4.0，DM OpenAPI 在 v6.0.0 版本中有较大变化。
```
Comment (Ehco1996) [inline dm/dm-open-api.md:30]: 有很多 breaking change，不是很好描述

File: dm/dm-open-api.md [hunk 15]
```diff
@@ -5,17 +5,29 @@ summary: 了解如何使用 OpenAPI 接口来管理集群状态和数据同步
...
 openapi = true
 ```
 
+如需开启 OpenAPI，并且是通过 tiup 部署的 DM 集群则需要在拓扑文件中添加如下配置：
+
+```yaml
+server_configs:
+  master:
+    openapi: true
+```
```
Comment (TomShawn) [inline dm/dm-open-api.md:22]:
````suggestion
如需开启 OpenAPI，可通过以下方法：

+ 如果你的 DM 集群是通过二进制直接部署的，则在 DM-master 的配置文件中添加如下配置：

    ```toml
    openapi = true
    ```

+ 如果你的 DM 集群是通过 TiUP 部署的，则在拓扑文件中添加如下配置：

    ```yaml
    server_configs:
      master:
        openapi: true
    ```
````

File: dm/dm-open-api.md (hunk 14 above)
Comment (TomShawn) [reply dm/dm-open-api.md:30]: 如果不详说的话，个人觉得这句话没有太大意义。也许在 release notes 中详说打破的兼容性会更好。

File: dm/dm-open-api.md [hunk 16]
```diff
@@ -716,133 +907,364 @@ curl -X 'POST' \
...
+}
+```
+
+## 删除同步任务
+
+该接口是一个同步接口，请求成功后返回的 Status Code 是 204。
+
+### 请求 URI
+
+ `DELETE /api/v1/tasks/{task-name}`
```
Comment (TomShawn) [inline dm/dm-open-api.md:1023]:
```suggestion
`DELETE /api/v1/tasks/{task-name}`
```

File: dm/dm-open-api.md [hunk 17]
```diff
@@ -716,133 +907,364 @@ curl -X 'POST' \
...
+
+该接口是一个同步接口，请求成功会返回对应同步任务的信息。
+
+> **注意：**
+>
+> 更新同步任务配置时，须确保该任务处于暂停状态，并已经运行到增量同步的阶段，且仅有部分字段可以更新。
+
+### 请求 URI
+
+ `PUT /api/v1/tasks/{task-name}`
```
Comment (TomShawn) [inline dm/dm-open-api.md:1045]:
```suggestion
`PUT /api/v1/tasks/{task-name}`
```

File: dm/dm-open-api.md [hunk 18]
```diff
@@ -716,133 +907,364 @@ curl -X 'POST' \
...
 }
 ```
 
+## 开始同步任务
+
+这是一个异步接口，请求成功的 Status Code 是 200。可通过[获取同步任务状态](#获取同步任务状态)接口获取最新的任务状态。
+
+### 请求 URI
+
+ `POST /api/v1/tasks/{task-name}/start`
```
Comment (TomShawn) [inline dm/dm-open-api.md:1241]:
```suggestion
`POST /api/v1/tasks/{task-name}/start`
```

File: dm/dm-open-api.md [hunk 19]
```diff
@@ -917,48 +1339,148 @@ curl -X 'GET' \
...
-## 暂停同步任务
+## 获取同步任务列表
 
-这是一个异步接口，请求成功的 Status Code 是 200，可通过[获取同步任务状态](#获取同步任务状态)接口获取最新的状态。
+该接口是一个同步接口，请求成功会返回对应的同步任务信息。
 
 ### 请求 URI
 
- `POST /api/v1/tasks/task-1/pause`
+ `GET /api/v1/tasks`
```
Comment (TomShawn) [inline dm/dm-open-api.md:1348]:
```suggestion
`GET /api/v1/tasks`
```

File: dm/dm-open-api.md [hunk 20]
```diff
@@ -917,48 +1339,148 @@ curl -X 'GET' \
...
-## 恢复同步任务
+## 获取同步任务的同步规则列表
 
-这是一个异步接口，请求成功的 Status Code 是 200，可通过[获取同步任务状态](#获取同步任务状态)接口获取最新的状态。
+该接口是一个同步接口，请求成功会返回对应同步任务的同步列表。
 
 ### 请求 URI
 
- `POST /api/v1/tasks/task-1/resume`
+ `GET /api/v1/tasks/{task-name}/sources/{source-name}/migrate_targets`
```
Comment (TomShawn) [inline dm/dm-open-api.md:1460]:
```suggestion
`GET /api/v1/tasks/{task-name}/sources/{source-name}/migrate_targets`
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

**Pull request merged**